lib/money/money.rb

```diff
@@ -211,11 +216,11 @@ def to_liquid
   end
 
   def to_json(options = {})
-    to_s
+    JSON.dump({value: value, currency: currency})
   end
 
   def as_json(*args)
-    to_s
+    to_json
```

Comment:
This should have the actual implementation.

Reply:
this was always wrong too, `as_json` needs to return a hash, not a string... fixed now